<launch>
    <param name="use_sim_time" value="true"/>
    <arg name="bag_name" default=""/>
    <node pkg="rosbag" type="play" name="player" output="screen" args="--pause --clock $(find db_config)/bags/$(arg bag_name) --loop"/>

    <group ns="dodobot" >
        <node name="db_tensorflow" pkg="db_tensorflow" type="db_tensorflow_node.py" output="screen" required="true">
            <param name="image_topic"          value="/camera/color/image_raw"/>
            <param name="depth_topic"          value="/camera/aligned_depth_to_color/image_raw"/>
            <param name="camera_info_topic"    value="/camera/color/camera_info"/>
            <param name="min_score_threshold"  value="0.7"/>
            <param name="max_boxes_to_draw"  value="30"/>
            <param name="bounding_box_border_px"  value="50"/>
            <param name="min_valid_z"  value="0.05"/>
            <param name="max_valid_z"  value="1.0"/>
            <param name="publish_in_robot_frame"  value="false"/>

            <param name="labels_path" value="$(find db_tensorflow)/annotations/label_map.pbtxt"/>
            <param name="model_path"  value="$(find db_tensorflow)/models/dodobot_objects_ssd_mobilenet_v2/saved_model"/>
            <!-- <param name="model_path"  value="$(find db_tensorflow)/models/dodobot_objects_ssd_resnet50_v1_fpn/saved_model"/> -->

            <rosparam param="marker_colors" file="$(find db_tensorflow)/config/marker_colors.yaml" command="load"/>
            <rosparam param="z_size_estimations" file="$(find db_tensorflow)/config/depth_estimations.yaml" command="load"/>
        </node>
    </group>
</launch>
